ui/webui/src/InstallationLanguage.jsx

```diff
@@ -39,105 +37,112 @@ const getLanguageNativeName = lang => lang["native-name"].v;
-            Promise.all(languages.map(lang => localizationProxy.GetLocales(lang))).then(res => {
-                return Promise.all(
-                    res.map((langLocales) => {
-                        return Promise.all(langLocales.map(locale => localizationProxy.GetLocaleData(locale)));
+            Promise.all(languages.map(lang => call("GetLocales", [lang])))
+                    .then(res => {
+                        return Promise.all(
+                            res.map(langLocales => {
+                                return Promise.all(langLocales.map(locale =>
+                                    call("GetLocaleData", [locale])
+                                ));
+                            })
+                        );
                     })
-                );
-            })
-                    .then(setLocales);
+                    .then(res => this.setState({ locales: res }, this.updateDefaultSelection));
         });
-    });
-
-    useEffect(() => {
-        // Once the component state contains the locale data from the API set the default selected language
-        if (selectedItem || !locales.length) {
-            return;
-        }
-
-        const languageId = lang.split("_")[0];
-        const currentLangLocales = locales.find(langLocales => getLanguageId(langLocales[0]) === languageId);
-        const currentLocale = currentLangLocales.find(locale => getLocaleId(locale) === lang);
-
-        setSelectedItem(getLocaleNativeName(currentLocale));
-    }, [locales, lang, selectedItem]);
-
-    const handleOnSelect = (event, lang) => {
-        onSelectLang(lang.localeId);
-        setSelectedItem(lang);
-    };
-
-    const isLoading = languages.length !== locales.length;
-    const options = (
-        !isLoading
-            ? locales.map(langLocales => {
-                const currentLang = languages.find(lang => getLanguageId(lang) === getLanguageId(langLocales[0]));
-
-                return (
-                    <SelectGroup
-                      label={cockpit.format("$0 ($1)", getLanguageNativeName(currentLang), getLanguageEnglishName(currentLang))}
-                      key={getLanguageId(currentLang)}>
-                        {langLocales.map(locale => (
-                            <SelectOption
-                              id={getLocaleId(locale).split(".UTF-8")[0]}
-                              key={getLocaleId(locale)}
-                              value={{
-                                  toString: () => getLocaleNativeName(locale),
-                                  // Add a compareTo for custom filtering - filter also by english name
-                                  localeId: getLocaleId(locale)
-                              }}
-                            />
-                        ))}
-                    </SelectGroup>
-                );
-            })
-            : []
-    );
+    }
+
+    updateDefaultSelection () {
+        const languageId = this.props.lang.split("_")[0];
```

Comment:
I think it would be good to mention in what format are the language strings or point to where it is defined/documented.